GVFS/GVFS.FunctionalTests/Tools/GitHelpers.cs

```diff
@@ -18,7 +19,7 @@ public static class GitHelpers
         /// GVFS.FunctionalTests.LockHolder program.
         /// </summary>
         private const string LockHolderCommandName = @"GVFS.FunctionalTests.LockHolder";
-        private const string LockHolderCommand = @"GVFS.FunctionalTests.LockHolder.exe";
+        private const string LockHolderCommand = @"GVFS.FunctionalTests.LockHolder.dll";

```

Comment:
Aren't we going to run a `dotnet publish` for this which will make it a `.exe` on Windows and no extension on Mac?

Reply:
(I'l jump in here since this was my recommendation in offline discussions with @alameenshah)

The Windows build does not currently have a `dotnet publish` step, and so it was simpler to have the tests call `dotnet` for the lock holder (like we do for the [unit tests](https://github.com/Microsoft/VFSForGit/blob/8c8927ff421d8181a12b7c53de76c1defd80092c/Scripts/RunUnitTests.bat#L9) and [functional tests](https://github.com/Microsoft/VFSForGit/blob/8c8927ff421d8181a12b7c53de76c1defd80092c/Scripts/RunFunctionalTests.bat#L42) on Windows).

We *do* have a publish step on Mac (and could launch the lock holder there without `dotnet`) but we can reduce the amount of Mac and Windows specific code by using `dotnet` on both.

CC: @nickgra for his thoughts on this approach

Reply:
the extension would be `.dll` on both Mac and Windows. Even after Publish.

Reply:
Are there any gotchas or potential differences in running one way or the other?

Reply:
In my mind, the risk of running the publish binaries has always been that if you're iterating in the IDE on some changes, you're only updating the BuildOutput version of the binaries, not the ones that live in Publish. We could publish as part of every build, but that'd slow down our inner loop.

Maybe that's a risk that is imagined, but I can see myself making that mistake at least :)